### ISSUE TITLE
Error when trying to view Storage details for host

### ISSUE THREAD

**Issue by beatcracker** [rancher/rancher#6513]
### Details
- Rancher Version: **v1.1.4**
- Cattle version: **v0.165.8.4**
- User Interface version: **v1.1.14**
- Docker version:
```yml
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        
 OS/Arch:      linux/amd64
```
- OS and where are the hosts located: **CentOS Linux 7 Core (3.10.0-327.36.3.el7.x86_64) in Google Compute Engine VM**
- Setup Details: **Single rancher node with one host, external MySQL DB (Cloud SQL), Convoy NFS storage pool**
- Environment Type: **Cattle**

### Steps to Reproduce

In the UI, go to the `Infrastructure` -> `Hosts` - Click on host -> Click on the `Storage` tab.

#### Error screenshot

![e.forEach is not a function](https://cloud.githubusercontent.com/assets/7583050/20017855/71996902-a2d6-11e6-859a-3586830e0397.png)

#### JS console log:

```
Application Error TypeError: e.forEach is not a function
    at Array.addObjects (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:14:20536)
    at t (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:55:22832)
    at n.followLink (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:55:8449)
    at http://1.2.3.4:8080/assets/ui-399966b411601a514e01ca71c9068e2e.js:79:6812
    at http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:14:13842
    at n.forEach (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:14:13622)
    at n.map (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:14:13811)
    at http://1.2.3.4:8080/assets/ui-399966b411601a514e01ca71c9068e2e.js:79:6787
    at y (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:19:15564)
    at b (http://1.2.3.4:8080/assets/vendor-ebfa7dda04b9f2b654291bf27e70bf22.js:19:15659)
```

#### Request (HAR)

```json
{
  "log": {
"version": "1.2",
"creator": {
  "name": "WebInspector",
  "version": "537.36"
},
"pages": [],
"entries": [
  {
"startedDateTime": "2016-11-04T18:32:31.551Z",
"time": 109.5000000204891,
"request": {
  "method": "GET",
  "url": "http://4.5.6.7:8080/v1/projects/1a5/hosts/1h1/storagepools",
  "httpVersion": "HTTP/1.1",
  "headers": [
{
  "name": "Pragma",
  "value": "no-cache"
},
{
  "name": "DNT",
  "value": "1"
},
{
  "name": "Accept-Encoding",
  "value": "gzip, deflate, sdch"
},
{
  "name": "Host",
  "value": "4.5.6.7:8080"
},
{
  "name": "Accept-Language",
  "value": "en-US,en;q=0.8,ru;q=0.6"
},
{
  "name": "User-Agent",
  "value": "Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.71 Safari/537.36"
},
{
  "name": "x-api-no-challenge",
  "value": "true"
},
{
  "name": "Accept",
  "value": "application/json"
},
{
  "name": "Cache-Control",
  "value": "no-cache"
},
{
  "name": "X-Requested-With",
  "value": "XMLHttpRequest"
},
{
  "name": "Cookie",
  "value": "PL=rancher; token=XXjPY25yVspkvZzMNCBdFz68aD4i7xX1ayuPNLjXX; CSRF=20009731011CAFCF57A09135A01E8266"
},
{
  "name": "Connection",
  "value": "keep-alive"
},
{
  "name": "Referer",
  "value": "http://4.5.6.7:8080/env/1a5/infra/hosts/1h1/containers"
},
{
  "name": "x-api-csrf",
  "value": "20009731011CAFCF57A09135A01E8266"
}
  ],
  "queryString": [],
  "cookies": [
{
  "name": "PL",
  "value": "rancher",
  "expires": null,
  "httpOnly": false,
  "secure": false
},
{
  "name": "token",
  "value": "XXjPY25yVspkvZzMNCBdFz68aD4i7xX1ayuPNLXX",
  "expires": null,
  "httpOnly": false,
  "secure": false
},
{
  "name": "CSRF",
  "value": "20009731011CAFCF57A09135A01E8266",
  "expires": null,
  "httpOnly": false,
  "secure": false
}
  ],
  "headersSize": 676,
  "bodySize": 0
},
"response": {
  "status": 200,
  "statusText": "OK",
  "httpVersion": "HTTP/1.1",
  "headers": [
{
  "name": "Date",
  "value": "Fri, 04 Nov 2016 18:33:23 GMT"
},
{
  "name": "Content-Encoding",
  "value": "gzip"
},
{
  "name": "Server",
  "value": "Jetty(9.2.11.v20150529)"
},
{
  "name": "X-Api-Account-Id",
  "value": "1a5"
},
{
  "name": "Vary",
  "value": "Accept-Encoding, User-Agent"
},
{
  "name": "Content-Type",
  "value": "application/json; charset=utf-8"
},
{
  "name": "X-Api-Client-Ip",
  "value": "9.8.7.6"
},
{
  "name": "X-Api-Schemas",
  "value": "http://4.5.6.7:8080/v1/projects/1a5/schemas"
},
{
  "name": "Set-Cookie",
  "value": "CSRF=20009731011CAFCF57A09135A01E8266;Path=/"
},
{
  "name": "Content-Length",
  "value": "866"
},
{
  "name": "Expires",
  "value": "Thu, 01 Jan 1970 00:00:00 GMT"
}
  ],
  "cookies": [
{
  "name": "CSRF",
  "value": "20009731011CAFCF57A09135A01E8266",
  "path": "/",
  "expires": null,
  "httpOnly": false,
  "secure": false
}
  ],
  "content": {
"size": 4661,
"mimeType": "application/json",
"compression": 3795
  },
  "redirectURL": "",
  "headersSize": 437,
  "bodySize": 866,
  "_transferSize": 1303
},
"cache": {},
"timings": {
  "blocked": 3,
  "dns": -1,
  "connect": -1,
  "send": 0,
  "wait": 105,
  "receive": 1.5000000204890966,
  "ssl": -1
},
"serverIPAddress": "4.5.6.7",
"connection": "322710"
  }
]
  }
}
```

#### Responce

```json
{
  "type": "collection",
  "resourceType": "storagePool",
  "links": {
"self": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools"
  },
  "createTypes": {
"registry": "http://1.2.3.4:8080/v1/projects/1a5/registries"
  },
  "actions": {},
  "data": [
{
  "id": "1sp1",
  "type": "storagePool",
  "links": {
"self": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1",
"account": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/account",
"credentials": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/credentials",
"hosts": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/hosts",
"images": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/images",
"volumes": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/volumes"
  },
  "actions": {
"update": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/?action=update",
"deactivate": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp1/?action=deactivate"
  },
  "name": "rancher-host-XX Storage Pool",
  "state": "active",
  "accountId": "1a5",
  "blockDevicePath": null,
  "created": "2016-11-02T17:39:23Z",
  "createdTS": 1478108363000,
  "description": null,
  "driverName": null,
  "externalId": null,
  "kind": "docker",
  "removed": null,
  "transitioning": "no",
  "transitioningMessage": null,
  "transitioningProgress": null,
  "uuid": "6b990cf6-718f-405b-841e-836c39648a7a",
  "volumeAccessMode": null,
  "volumeCapabilities": null
},
{
  "id": "1sp3",
  "type": "storagePool",
  "links": {
"self": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3",
"account": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/account",
"credentials": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/credentials",
"hosts": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/hosts",
"images": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/images",
"volumes": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/volumes"
  },
  "actions": {
"update": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/?action=update",
"deactivate": "http://1.2.3.4:8080/v1/projects/1a5/storagepools/1sp3/?action=deactivate"
  },
  "name": "convoy-nfs",
  "state": "active",
  "accountId": "1a5",
  "blockDevicePath": null,
  "created": "2016-11-04T00:58:58Z",
  "createdTS": 1478221138000,
  "description": null,
  "driverName": "convoy-nfs",
  "externalId": "convoy-nfs",
  "kind": "storagePool",
  "removed": null,
  "transitioning": "no",
  "transitioningMessage": null,
  "transitioningProgress": null,
  "uuid": "0ed7a651-e166-4817-889a-b2ae37a3c943",
  "volumeAccessMode": null,
  "volumeCapabilities": null
}
  ],
  "sortLinks": {
"accountId": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=accountId",
"agentId": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=agentId",
"created": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=created",
"description": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=description",
"driverName": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=driverName",
"external": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=external",
"externalId": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=externalId",
"id": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=id",
"kind": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=kind",
"name": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=name",
"physicalTotalSizeMb": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=physicalTotalSizeMb",
"removeTime": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=removeTime",
"removed": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=removed",
"state": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=state",
"uuid": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=uuid",
"virtualTotalSizeMb": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=virtualTotalSizeMb",
"volumeAccessMode": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=volumeAccessMode",
"zoneId": "http://1.2.3.4:8080/v1/projects/1a5/hosts/1h1/storagepools?sort=zoneId"
  },
  "pagination": {
"first": null,
"previous": null,
"next": null,
"limit": 100,
"total": null,
"partial": false
  },
  "sort": null,
  "filters": {
"accountId": null,
"agentId": null,
"created": null,
"description": null,
"driverName": null,
"external": null,
"externalId": null,
"id": null,
"kind": null,
"name": null,
"physicalTotalSizeMb": null,
"removeTime": null,
"removed": null,
"state": null,
"uuid": null,
"virtualTotalSizeMb": null,
"volumeAccessMode": null,
"zoneId": null
  },
  "createDefaults": {}
}
```

**Comment by tfiduccia** [comment]
This is a duplicate of #5982.  This has been fixed and will be in our 1.2.0 release.

**Comment by beatcracker** [comment]
Thanks! Not sure how I've managed to miss this issue, because I did search for this error. Probably not my day today.
